Observer/Order/AfterSave.php

```diff
@@ -150,7 +137,7 @@ protected function isOrderNew($order)
     protected function isSameState($order)
     {
         $oldData = $this->registry->registry(self::REGISTRY_KEY_ORDER_OLD_DATA);
-        $oldValue = $oldData['state'];
+        $oldValue = $oldData === null ? null : $oldData['state'];
```

Comment:
PHP 8 now throws a warning (which Magento helpfully converts into an exception) when we access null in this way. We were skating by with some ugly code previously.

Reply:
```suggestion
        $oldValue = $oldData['state'] ?? null;
```